iocore/net/quic/QUICStream.cc

```diff
@@ -599,7 +598,9 @@ QUICStream::_on_frame_acked(QUICFrameInformationUPtr &info)
     break;
   }
 
-  this->_state.update_on_ack();
+  if (this->_is_reset_complete || this->_is_transfer_complete) {
+    this->_state.update_on_transport_send_event();
```

Comment:
This is confusing. We are calling `update_on_transport_send_event` in `_on_frame_acked`. Where's "send event"?

Reply:
may be send_complete_event ?

Reply:
I don't think we need `update_on_transport_send_event` because `QUICTransferProgressProvider` provides enough information and we can check it in `update_on_ack`.

Reply:
Yep, lets inherit this code in Provider.